benchmark/module/require.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+var common = require('../common.js');
+
+var bench = common.createBenchmark(main, {
+  n: [512]
+});
+
+function main(opts) {
+  var n = opts.n;
+  bench.start();
+  for (var i = 0; i < n; ++i) {
+    require('os');
+    require('../common.js');
```

Comment:
The cache needs to be clean after requiring :)

Reply:
clear cache after require:
src/js/module.js:268
```js
  iotjs_module_t.cache = {};
  return module.exports;
```

before:
```js
module/require.js
module/require.js n=512: 239.6341087450955
```

after:
```js
module/require.js
module/require.js n=512: 231.75294154016615

module/require.resolve.js
module/require.resolve.js n=512: 476.3931267019459
```

Reply:
Could clear cache at your test code?

Reply:
I wanted to, but I found there is not cache interface exposed before I add resolve method.

Reply:
@Txiaozhe #458 has been merged that added the `require.cache` function, so you could rebase your PR and clear cache in your bench script :)

Reply:
That is a good idea :) I will do it!

Reply:
clear cache:
/benchmark/module/require.resolve.js
```js
function main(opts) {
  var n = opts.n;
  bench.start();
  for (var i = 0; i < n; ++i) {
    require.resolve('os');
    require.resolve('../common.js');
    require.cache = {};
  }
  bench.end(n);
}
```
/benchmark/module/require.js
```js
function main(opts) {
  var n = opts.n;
  bench.start();
  for (var i = 0; i < n; ++i) {
    require('os');
    require('../common.js');
    require.cache = {};
  }
  bench.end(n);
}
```

before:
```shell
module/require.js
module/require.js n=512: 459.0993043192024
```
after:
```shell
module/require.js
module/require.js n=512: 511.4249267068896

module/require.resolve.js
module/require.resolve.js n=512: 477.2248819830091
```

Reply:
It seem too much commit here, need I recommit it?